src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -594,12 +666,12 @@ public void gatherXp()
      */
     public void addExperience(final double xp)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)